drone/src/run.rs

```diff
@@ -10,26 +10,38 @@ use dis_spawner::cli::init_cli;
 use dis_spawner::logging::TracingHandle;
 use dis_spawner::messages::logging::Component;
 use dis_spawner::retry::do_with_retry;
+use dis_spawner::types::DroneId;
 use dis_spawner::NeverResult;
 use futures::future::try_join_all;
 use futures::Future;
 use signal_hook::{consts::SIGINT, iterator::Signals};
 use std::{pin::Pin, thread};
 
 async fn drone_main() -> NeverResult {
-    let config: DroneConfig = init_cli()?;
-    let plan = DronePlan::from_drone_config(config).await?;
+    let mut config: DroneConfig = init_cli()?;
+
+    // Extract drone ID, or generate one if necessary.
+    // DronePlan::from_drone_config will do this if we don't do it here,
+    // but we want to initialize tracing now, so we ensure the value
+    // exists before calling from_drone_config.
```

Comment:
Should we remove that logic from `from_drone_config` so that it's clear that `drone_id` is always determined here?

Reply:
I think not, because then it becomes part of the unwritten contract we have with `from_drone_config` if we (or someone else) ever calls it from elsewhere.